### ISSUE TITLE
Anima Fast 在 Linux 下安装和审计配置不匹配

### ISSUE THREAD

**Issue by Abiogenesisss** [wochenlong/lora-scripts-next#77]
## 问题

Linux 环境下安装 Anima Fast 时，安装器和审计配置有几处不匹配，导致插件一直无法变成 ready。

环境大致是：

- Linux
- Python 3.13.13
- CUDA/cu130
- Anima source commit: `87819818975e08167cda8a6f615776e46e889f80`

## 遇到的报错

一开始会因为 `flash-attn` 解析失败：

```text
there is no version of flash-attn==2.8.3+cu130torch2.11
```

手动给 `anima-overrides-cu130.txt` 加 direct wheel URL 后，这一步可以过。

之后又遇到 constraints 冲突：

```text
anima-lora==0.1.0 depends on torchvision{sys_platform == 'linux'}>=0.27.0,<0.28
and torchvision{sys_platform == 'linux'}==0.26.0+cu130
```

把 Anima Fast 的 constraints 里 `torchvision` 改到 `0.27.0+cu130` 后可以继续。

最后 audit 又在 Linux 上检查了 `triton-windows`：

```text
anima: triton-windows expected 3.7.0.post26, got None
```

但这个包应该只在 Windows 下需要。`anima_lora` 的 metadata 里也是这样声明的：

```text
Requires-Dist: triton-windows>=3.6.0; sys_platform == "win32"
```

## 临时绕过方式

我这边最后是这样绕过的：

- `anima-constraints-cu130.txt` 里改成：

```text
torch==2.12.0+cu130
torchvision==0.27.0+cu130
```

- 删除 constraints 里的：

```text
triton-windows==3.7.0.post26
```

- `anima-overrides-cu130.txt` 里添加：

```text
flash-attn @ https://github.com/mjun0812/flash-attention-prebuild-wheels/releases/download/v0.9.4/flash_attn-2.8.3%2Bcu130torch2.11-cp313-cp313-linux_x86_64.whl
```

- `environment.py` 里把 `triton-windows` 的 audit 改成只在 Windows 下检查。

这样之后 Linux 环境可以继续安装/审计。

## 建议

Linux 下应该跳过 `triton-windows` 的安装和审计；另外 Anima Fast 的 constraints 可能需要和当前 `anima_lora` 的 Linux 依赖同步一下。

**Comment by wochenlong** [comment]
正在关注此问题

**Comment by MikumikuDAIFans** [comment]
阶段性证据已回填，本 issue 暂不关闭，等 Phase 6 Linux/Fast 真实 smoke 后再最终关闭。

- commit：`96a28ec` 当前签收基线，安装/审计路径已有平台化测试覆盖。
- 当前验证结果：Windows/portable RC 上 seeded Fast install/audit/ready、dry-run、API training smoke 均通过；平台化 install/audit 单元测试通过。
- 证据路径：
  - `E:\OpenSourceTeamWork\Real_test\release_candidate\logs\portable-rc-fast-install-seeded-ready-v2-20260604.txt`
  - `E:\OpenSourceTeamWork\Real_test\release_candidate\logs\portable-rc-fast-dry-run-after-seeded-ready-20260604.txt`
  - `E:\OpenSourceTeamWork\Real_test\release_candidate\logs\portable-rc-fast-api-train-smoke-v2-20260604.txt`
  - `E:\OpenSourceTeamWork\Real_test\release_candidate\logs\pytest-phase2-fast-portable-focused-20260604.txt`
- 残余边界：issue 原题是 Linux 下安装/审计配置不匹配；任务书已把 Linux/WSL2 clean env 的 install -> audit -> ready -> dry-run -> train 放入 Phase 6。该真实 Linux smoke 完成前，本 issue 保持 open。

**Comment by MikumikuDAIFans** [comment]
Phase 6 Linux/Fast real smoke is now signed and pushed, so this issue can be closed.

Commit / branch:
- `be6658d3ac69d3e5b9334e216e64766d5020774e` on `anima_lora`
- Commit message: `Fix Linux Anima Fast environment signoff`

What changed:
- Linux cu130 constraints now align with Anima source requirements:
  - `torch==2.12.0+cu130 ; sys_platform == "linux"`
  - `torchvision==0.27.0+cu130 ; sys_platform == "linux"`
- `triton-windows==3.7.0.post26` is scoped to Windows only.
- Linux audit no longer requires `triton-windows`.
- Linux install copy localizes the Anima source Linux `flash-attn` dependency to the cu130 wheel.
- Linux/WSL headless runtime is covered with `opencv-python-headless` compatibility.

Real Linux/WSL2 signoff:
- Evidence report: `E:\OpenSourceTeamWork\Real_test\linux_fast_smoke\final-report.md`
- Environment: WSL2 `kali-linux`, kernel `6.6.114.1-microsoft-standard-WSL2`
- GPU: `NVIDIA GeForce RTX 4060 Ti`
- Main controller runtime: Python `3.12.13`, `torch==2.12.0+cu130`, `torch.cuda.is_available=true`
- Anima Fast runtime: Python `3.13.13`, `torch==2.12.0+cu130`, `torchvision==0.27.0+cu130`, `flash-attn==2.8.3+cu130torch2.11`, `triton==3.7.0`, `triton-windows=None`
- Flow completed: install -> audit -> ready -> dry-run/preflight -> preprocess -> train
- Preflight result: `ok=true`, `errors=[]`
- Training data/model: real Anima base model and small real dataset from local resources
- Training completed: 4/4 steps on `cuda:0` with bf16
- Main output SHA256: `724f4bbaa411093228f82a0879b235fb159a8c5b681a282800531e310f806f51`

Validation:

```powershell
E:\OpenSourceTeamWork\lora-scripts-next\venv\Scripts\python.exe -m pytest -q tests/test_anima_fast_environment_installer.py tests/test_anima_fast_backend.py tests/test_anima_fast_plugin_api.py
```

Result:

```text
51 passed, 3 warnings, 2 subtests passed
```

Residual boundaries:
- This was signed on WSL2 Linux, not a separate cloud AutoDL host.
- Training smoke used `attn_mode=torch`; `flash-attn` import/preflight is verified, but flash attention training mode is outside this issue closure.
- Formal public release / portable archive publishing is tracked separately.

**Comment by MikumikuDAIFans** [comment]
按团队流程要求，先重新打开此 issue：此前关闭时附上的本地验证/Real_test 证据继续保留，但在团队正式合并代码之前，不再将这些 issue 标记为 closed。

后续等对应代码正式合并并完成最终发布/验收口径确认后，再统一关闭。
